src/standalone/userJupyterServer/userServerUrlProvider.ts

```diff
@@ -286,7 +298,10 @@ export class UserJupyterServerUrlProvider implements IExtensionSyncActivationSer
                     }
                 }),
                 input.onDidHide(() => {
-                    resolve(undefined);
+                    inputWasHidden = true;
```

Comment:
As soon as the password input is displayed this gets hidden
Then we go back to the multi-select quick pick and the password input gets dismissed